### DISCUSSION TITLE
Have to insert a field even though its value is allowed to be NULL.

### DISCUSSION THREAD

**Discussion by vonkoff** [planetscale/discussion#288] [Bug Reports] [answered]
I used `Describe` to check the table in my database. This can be seen with the second image below.

As can be seen, the field `postPaid`  can be `NULL`. But, for some reason when I try to insert data without it (as seen in my code and the first image) it throws an error.

I really am unsure as to why this could be happening. 

```sql
INSERT INTO Search (job, url, description, addressId, guildId, employerId, postPaid)
VALUES ('sanitation engineer', 'yurt.com', 'fake descripiton', 1, 1,1, 'test')
;

Completed in 44 ms
INSERT INTO Search (job, url, description, addressId, guildId, employerId)
VALUES ('sanitation engineer', 'y.com', 'fake descripiton', 1, 1,1)
;

target: goodunionjobs.-.primary: vttablet: rpc error: code = Unknown desc = Field 'postPaid' doesn't have a default value (errno 1364) (sqlstate HY000) (CallerID: planetscale-admin): Sql: "insert into Search(job, url, description, addressId, guildId, employerId) values (:vtg1, :vtg2, :vtg3, :vtg4, :vtg5, :vtg6)", BindVars: {vtg1: "type:VARCHAR value:\"sanitation engineer\""vtg2: "type:VARCHAR value:\"y.com\""vtg3: "type:VARCHAR value:\"fake descripiton\""vtg4: "type:INT64 value:\"1\""vtg5: "type:INT64 value:\"1\""vtg6: "type:INT64 value:\"1\""}
```
```sql
CREATE TABLE `Search` (
	`id` int NOT NULL AUTO_INCREMENT,
	`job` text NOT NULL,
	`url` varchar(500) NOT NULL,
	`description` text NOT NULL,
	`salary` text,
	`benefits` json,
	`payType` varchar(10),
	`employmentType` varchar(20),
	`color` smallint,
	`postPaid` varchar(10),
	`addressId` int NOT NULL,
	`guildId` int NOT NULL,
	`employerId` int NOT NULL,
	`employerImageId` int,
	PRIMARY KEY (`id`),
	UNIQUE KEY `Search_url_key` (`url`)
) ENGINE InnoDB,
  CHARSET utf8mb4,
  COLLATE utf8mb4_unicode_ci;
```
![Screen Shot 2022-09-21 at 9 08 55 PM](https://user-images.githubusercontent.com/16548992/191636567-e49f966d-241e-4cd3-afdb-0e378c482bba.png)
![Screen Shot 2022-09-21 at 9 09 01 PM](https://user-images.githubusercontent.com/16548992/191636554-761a359b-03ff-4df2-9bd1-413a804af378.png)

**Comment by orware** [comment, marked as answer]
Hello @vonkoff,

Thank you for the report! It is a bit puzzling behavior since I tested out adding the same table into a test database and wasn't able to recreate the same issue.

First I added in the same table using the example `CREATE TABLE` query you shared above and then ran the same two `INSERT` queries afterward without any difficulty:
![image](https://user-images.githubusercontent.com/931650/191658042-784e15b7-f881-4282-80c2-8ada901e9a6a.png)

And I also used `DESCRIBE` to compare the output but it looks the same to me:
![image](https://user-images.githubusercontent.com/931650/191657984-b4cc507b-f6d7-4021-8f40-d06e9cac60b0.png)

If you are continuing to experience this issue, I would suggest a quick drop and recreation of the table, since there are only 14 rows, if that wouldn't be too much trouble to try first to see if the issue persists. 

If it does, please reach out to us using https://support.planetscale.com/ so that we may assist you more directly and see why this may be happening.

**Reply by vonkoff** [reply]
Weird. Dropping and adding the table back fixed the issue! Wonder what made this happen in the first case.
